### DISCUSSION TITLE
Using PowerDNS to create a network-wide ad and domain blocking service for multiple customers (tenants) : Need suggestions

### DISCUSSION THREAD

**Discussion by ksingh-scogo** [PowerDNS/pdns#14073] [General]
I want to build a solution that uses powerDNS to create a network-wide ad and domain blocking service for multiple customers (tenants).

Here's the breakdown:

- Multi-tenancy: I need powerDNS to support managing separate block lists for each customer. Ideally, a single powerDNS instance would handle everything.
- Customer assignment: I'm unsure how powerDNS would identify which customer's block list to apply, since everyone would use the same powerDNS server IP.
- Cost-effective solution: If multi-tenancy on a single server isn't feasible, I'd like to deploy powerDNS on Kubernetes with unique URLs for each customer (e.g., https://c1.mypowerDNS.example.com, https://c2.mypowerDNS.example.com).
However, I'm unsure how clients (devices and routers) would reach these URLs since DNS resolution typically works with IPs.
Community Help Needed:

I'm reaching out to the community for advice on:

- Can powerDNS handle multi-tenancy for my use case?
- If not, how can I achieve multi-tenancy with separate powerDNS instances on Kubernetes?
- What tools would I need to implement this solution with unique customer URLs?

Note :

1) The router on each customer side will be an OpenWRT based Router for which we will be having full access.
2) If this feature does not exists currently and needs be custom built, we would be willing to sponsor (in our limited capacity) the maintainers of this project and support development of this feature and get this into opensource community for others to use.

**Comment by omoerbeek** [comment]
We build this and it is available as a (paid) product. https://www.powerdns.com/powerdns-protect
